test/attr/ignore_deprecation_warnings.swift

```diff
@@ -0,0 +1,71 @@
+// RUN: %target-typecheck-verify-swift
+
+@available(*, deprecated)
+func deprecatedFunc() {}
+
+@available(*, deprecated)
+func deprecatedFuncWithReturnValue() -> Int {
+    return 3
+}
+
+@available(*, deprecated)
+func deprecatedFuncWithReturnValue2() -> String {
+    return ""
+}
+
+@available(*, deprecated)
+var deprecatedVariable = ""
+
+@available(*, deprecated)
+struct deprecatedType {}
+
+
+@ignoreDeprecationWarnings 
+// FIXME: deprecation warnings are still diagnosed for top-level variable declaration
+// this has to do with Swift diagnosing deprecation warnings for deprecated top-level variables which are assigned to a deprecated property
+var topLevelDecl = deprecatedVariable // expected-warning {{'deprecatedVariable' is deprecated}}
+
+@ignoreDeprecationWarnings
+func x() {
+    deprecatedFunc()
+}
+
+let instanceThatSouldProduceWarning = deprecatedType() // expected-warning {{'deprecatedType' is deprecated}}
+
+var s = deprecatedVariable // expected-warning {{'deprecatedVariable' is deprecated}}
+
+func funcThatShouldProduceWarning() {
+    deprecatedFunc() // expected-warning {{'deprecatedFunc()' is deprecated}}
+}
+
+struct A {
+    @ignoreDeprecationWarnings let x = deprecatedFuncWithReturnValue()
+    let shouldWarn = deprecatedFuncWithReturnValue() // expected-warning {{'deprecatedFuncWithReturnValue()' is deprecated}}
+}
+
+@ignoreDeprecationWarnings
+extension A {
+    func aFunc() -> Int {
+        deprecatedFuncWithReturnValue()
+    }
+}
+
+@ignoreDeprecationWarnings
+class anotherType {
+    func a() {
+        _ = deprecatedFuncWithReturnValue()
+    }
+
+    let shouldntWarn1 = deprecatedType()
+    let shouldntWarn2 = deprecatedFuncWithReturnValue2()
+}
+
+@inlinable
+@ignoreDeprecationWarnings
+func multipleAttrsFunc() -> deprecatedType {
+    return deprecatedType()
+}
```

Comment:
You're getting an additional error here because the function is marked `inlinable`, meaning that `deprecatedType` and its initializer is required to be public. If you just want to test multiple attributes, you could use `@discardableResult` instead